### PR TITLE
LXD storage with tmpfs

### PR DESCRIPTION
**Pull request by yhaliaw**

### CODE REVIEW

**Review by jdkandersson — DISMISSED** [review]

File: src/charm.py [hunk 1]
```diff
@@ -155,9 +157,29 @@ def __init__(self, *args, **kargs) -> None:
...
+        Args:
+            path: Path to directory for memory storage.
+            size: Size of the tmpfs in kilobytes.
+        """
+        tmpfs_path = Path("/var/snap/lxd/common/ram")
+
+        if not tmpfs_path.exists():
+            tmpfs_path.mkdir(parents=True)
+            path.mkdir(parents=True, exist_ok=True)
+            execute_command(
```
Comment (yanksyoon) [inline src/charm.py:172]: Should we be checking the result of these executions?

File: src/runner.py [hunk 2]
```diff
@@ -243,6 +244,25 @@ def _ensure_runner_profile(self) -> None:
         else:
             logger.info("Found existing runner LXD profile")
 
+    @retry(tries=5, delay=5, local_logger=logger)
+    def _ensure_runner_storage_pool(self) -> None:
+        """Ensure the runner storage pool exists."""
+        if not self._clients.lxd.storage_pools.exists("runner"):
```
Comment (yanksyoon) [inline src/runner.py:250]: Would it be better if the else came first and returned immediately?

File: src/charm.py [hunk 3]
```diff
@@ -166,15 +188,26 @@ def _get_runner_manager(
                 name.
 
         Returns:
-            A instance of RunnerManager if the token and path configuration can be found.
+            A instance of RunnerManager.
```
Comment (arturo-seijas) [inline src/charm.py:191]: s/A/An

**Review by yanksyoon — DISMISSED** [review]
Minor nitpick comments, LGTM!

File: src/runner_manager.py [hunk 4]
```diff
@@ -204,7 +208,7 @@ def update_runner_bin(self, binary: RunnerApplication) -> None:
         sha256 = hashlib.sha256()
 
         with RunnerManager.runner_bin_path.open(mode="wb") as file:
-            for chunk in response.iter_content(decode_unicode=False):
+            for chunk in response.iter_content(chunk_size=128 * 1024, decode_unicode=False):
```
Comment (arturo-seijas) [inline src/runner_manager.py:211]: Can you define these magic numbers in a constant?

**Review by jdkandersson — DISMISSED** [review]

**Review by jdkandersson — DISMISSED** [review]

**Review by jdkandersson — DISMISSED** [review]

**Review by arturo-seijas — DISMISSED** [review]

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by jdkandersson — APPROVED** [review]

**Pull request merged**